data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/special_damage_type.cfg

```diff
@@ -348,3 +348,160 @@
+                    [damage]
+                        value=12
+                    [/damage]
+                    [damage_type]
+                        alternative_type=cold
+                    [/damage_type]
+                    [chance_to_hit]
+                        value=100
+                    [/chance_to_hit]
+                [/set_specials]
+            [/effect]
+            [filter]
+                id=bob
+            [/filter]
+        [/object]
+        [object]
+            silent=yes
+            [effect]
+                apply_to=resistance
+                replace=yes
+                [resistance]
+                    cold=200
+                    blade=100
+                [/resistance]
+            [/effect]
+            [effect]
+                apply_to=attack
+                [set_specials]
+                    mode=append
+                    [attacks]
+                        value=1
+                    [/attacks]
+                    [damage]
+                        value=12
+                    [/damage]
+                    [damage_type]
+                        alternative_type=fire
+                    [/damage_type]
+                    [chance_to_hit]
+                        value=100
+                    [/chance_to_hit]
+                [/set_specials]
+            [/effect]
+            [filter]
+                id=alice
+            [/filter]
+        [/object]
+
+        [store_unit]
+            [filter]
+                id=alice
+            [/filter]
+            variable=a
+            kill=yes
+        [/store_unit]
+        [store_unit]
+            [filter]
+                id=bob
+            [/filter]
+            variable=b
+        [/store_unit]
+        [unstore_unit]
+            variable=a
+            find_vacant=yes
+            x,y=$b.x,$b.y
+        [/unstore_unit]
+        [store_unit]
+            [filter]
+                id=alice
+            [/filter]
+            variable=a
+        [/store_unit]
+
+        [do_command]
+            [attack]
+                weapon=0
+                defender_weapon=0
+                [source]
+                    x,y=$a.x,$a.y
+                [/source]
+                [destination]
+                    x,y=$b.x,$b.y
+                [/destination]
+            [/attack]
+        [/do_command]
+        [store_unit]
+            [filter]
+                id=alice
+            [/filter]
+            variable=a
+        [/store_unit]
+        [store_unit]
+            [filter]
+                id=bob
+            [/filter]
+            variable=b
+        [/store_unit]
+        #damage without modification are 12, if test fail hitpoints !=76
+        #if succed then damage by alice 24(bob more vulnerable to fire)
+        #if succed then damage by bob 24(alice vulnerable to cold, cold [damage] is used)
+        {ASSERT ({VARIABLE_CONDITIONAL a.hitpoints equals 76})}
+        {ASSERT ({VARIABLE_CONDITIONAL b.hitpoints equals 76})}
```

Comment:
This doesn't test issue #8631 - in #8631, it wants to check that an "only on defense" ability reduces the damage.

Reply:
The new one's closer (and I think it should be kept), but we need another one where Bob is by default vulnerable to the damage, but has an ability that reduces the damage, but still to a point where that attack would be used.

For example, an undead unit being attacked with a blade+arcane weapon, and a skill that reduces their arcane damage slightly.

Reply:
it done